### PR TITLE
Add Apache Hive 3.1.2

### PR DESCRIPTION
**Pull request by nmahadevuni**
Cherry pick of https://github.com/trinodb/docker-images/pull/79

Co-authored-by: Guy Cohen <guyc@gett.com>

On top of the original PR from trino, additional changes are made to
1) Run SOCKS proxy 
2) Add "hive.metastore.disallow.incompatible.col.type.changes" set to 'false' in hive-site.xml. This was changed from false to true from Hive versions < 3. It was causing AbstractTestHiveClient#testDropColumn to fail.
3) Added user hive. This is the user we run tests as.

### CODE REVIEW

**Comment by nmahadevuni** [comment]
@aweisberg : This PR contains changes to build Apache Hive 3.1.2 docker images. I get some errors while building images locally using the instructions in the README.md.  I'm using Docker For Mac 3.3.1 (Docker Engine V20.10.5). Specifically the errors relate to the "docker history" command run by bin/label-parent.sh which now doesn't get the actual IMAGE ID but an empty string.

**Comment by aweisberg** [comment]
For me it fails here https://pastebin.com/uAfuk4Bz
```
cd prestodb/centos7-oj8 && time /bin/sh -c "( tar -czh . | docker build   -t prestodb/centos7-oj8:latest --label com.teradata.git.hash=d5c4ce0c1ea22d21bdb6c48aa23cdae289c4ef6c - )"
[+] Building 0.1s (2/2) FINISHED                                                                                                        
 => [internal] load remote build context                                                                                           0.0s
 => ERROR copy /context /                                                                                                          0.1s
------
 > copy /context /:
------
failed to solve with frontend dockerfile.v0: failed to read dockerfile: Error processing tar file(gzip: invalid header): 
```

File: prestodb/hive3.1-hive/files/root/setup.sh
Comment (aweisberg) [inline prestodb/hive3.1-hive/files/root/setup.sh:23]: Missing newline

**Review by aweisberg — APPROVED** [review]
@highker Unfortunately automatically building and publishing to DH is broken. There are multiple levels of issues there. I built and uploaded to DH to unblock things.

File: Makefile
Comment (aweisberg) [inline Makefile:193]: This should be a separate commit/PR since it is unrelated to what you are doing here.

File: prestodb/centos7-oj8/files/usr/local/bin/apply-site-xml-override
Comment (aweisberg) [inline prestodb/centos7-oj8/files/usr/local/bin/apply-site-xml-override:22]: This should be in the initial commit since the initial commit should build (no broken revisions)

File: prestodb/hive3.1-hive/Dockerfile
Comment (aweisberg) [inline prestodb/hive3.1-hive/Dockerfile:13]: Same

File: prestodb/hive3.1-hive/Dockerfile
Comment (aweisberg) [inline prestodb/hive3.1-hive/Dockerfile:68]: This is an addition for debugging via SSH Is this safe? It looks like is is listening on all interfaces

**Comment by aweisberg** [comment]
I just have a question about the SSH daemon and some requests for cleanup and clarity.

Can you mention the PR description and second commit what the additional changes you are layering on top of the original PR are? Looks like you are trying to enable impersonation.

**Review by aweisberg — CHANGES_REQUESTED** [review]
I just have a question about the SSH daemon and some requests for cleanup and clarity.

Can you mention the PR description and second commit what the additional changes you are layering on top of the original PR are? Looks like you are trying to enable impersonation.

File: Makefile
Comment (nmahadevuni) [reply Makefile:193]: Will remove this.

File: prestodb/centos7-oj8/files/usr/local/bin/apply-site-xml-override
Comment (nmahadevuni) [reply prestodb/centos7-oj8/files/usr/local/bin/apply-site-xml-override:22]: Should I merge them in to a single commit and force push?

File: prestodb/hive3.1-hive/Dockerfile
Comment (nmahadevuni) [reply prestodb/hive3.1-hive/Dockerfile:68]: This is same as previous version image. SOCKS proxy is required to connect to services running on the container. All the hive integration tests configuration requires this to be setup. EXPOSE command here doesn't really have any effect though as it acts just as documentation. docker-compose.yml maps the actual ports required.

**Comment by nmahadevuni** [comment]
> I just have a question about the SSH daemon and some requests for cleanup and clarity.
> 
> Can you mention the PR description and second commit what the additional changes you are layering on top of the original PR are? Looks like you are trying to enable impersonation.

The core-site.xml impersonation changes were a mistake, not required. I will remove them.

File: prestodb/hive3.1-hive/Dockerfile
Comment (nmahadevuni) [reply prestodb/hive3.1-hive/Dockerfile:13]: The "unlabelled" tag is required to go through Makefile build process. I added latest since we are building manually. Should I revert this to unlabelled?

File: prestodb/centos7-oj8/files/usr/local/bin/apply-site-xml-override
Comment (aweisberg) [reply prestodb/centos7-oj8/files/usr/local/bin/apply-site-xml-override:22]: Yes please

File: prestodb/hive3.1-hive/Dockerfile
Comment (aweisberg) [reply prestodb/hive3.1-hive/Dockerfile:13]: Hmm OK, well there is no great answer, but yeah lets go with latest and assume next time the Makefile is used.

File: prestodb/hive3.1-hive/Dockerfile
Comment (aweisberg) [reply prestodb/hive3.1-hive/Dockerfile:68]: SGTM

File: prestodb/centos7-oj8/files/usr/local/bin/apply-site-xml-override
Comment (nmahadevuni) [reply prestodb/centos7-oj8/files/usr/local/bin/apply-site-xml-override:22]: Done.

File: prestodb/hive3.1-hive/files/root/setup.sh
Comment (nmahadevuni) [reply prestodb/hive3.1-hive/files/root/setup.sh]: Done

**Review by aweisberg — APPROVED** [review]

**Comment by aweisberg** [comment]
We should build and upload this before merging.

**Comment by aweisberg** [comment]
@highker can you merge this? The PR build doesn't work at all right now, but the image from this PR is built and uploaded.

**Comment by highker** [comment]
@aweisberg, @nmahadevuni, so do we want a rebase and force push?

**Review by highker — APPROVED** [review]

**Comment by nmahadevuni** [comment]
> @aweisberg, @nmahadevuni, so do we want a rebase and force push?

No PRs were merged after I created this one. I don't think a rebase is required.

**Pull request merged**
